### DISCUSSION TITLE
Develop 1 reusable components library with web components

### DISCUSSION THREAD

**Discussion by eddybrando** [Qiskit/qiskit.org#2326] [Ideas]
After being testing and developing our [reusable web components library](https://github.com/qiskit-community/qiskit-web-components) for a while now, I believe it’s time to make a decision in regards to the approach we want to take for the future.

Do we want to develop, maintain and use:

1. One library of **Vue components**
2. One library of **web components**
3. Two libraries (**Vue & web** components)

I believe the last approach (2 libraries) is clearly overkill. But if you’re for that option, kindly join the discussion as well.

There are a couple of things to consider:

1. Carbon itself is moving to support web components primarily and our library is based on / extends Carbon.
2. We have found no major issues with the web components approach, or those issues have workarounds.
3. Web components is a native standard.
4. Jupiter Hub uses React and we might need to integrate our components in the future with a Qiskit Jupiter Hub project, which would then need a re-do of the components for React, if we were developing them in Vue.
5. To support Vue 2 and 3 we need `vue-demi`, which is  nice, but it’s very fragile and specific in terms of dependencies and has caused already some issues when updating dependencies in the Textbook Beta project.

Considering this, I’d like to propose to make the decision to develop our reusable components library as web components.

How to exactly proceed will be discussed later, but it'll be progressive to avoid disruptions.

**Comment by techtolentino** [comment]
Thanks for opening this discussion Eddy.

Regarding 1 - we should get verification on this?, mainly because I haven't seen this announced or stated anywhere from their team.

**Reply by eddybrando** [reply]
That's true. Pure speculation here. It might be very useful to know this even if it's only a confirmation of the intent and not an official statement.

Focusing the efforts on the web components library instead of the Vue components one would mean that it would also be easier for us to extend the former than the latter.

**Comment by abdonrd** [comment]
Happily agree to use Web Components!

**Comment by korgan00** [comment]
I think that web components are the way. If for some reason, we need to do an implementation for other frameworks, it should be easy to create a wrapper. I think it covers most of the use cases for us, so agree with the migration.

**Comment by eddybrando** [comment]
Thank you all for your contributions.

We will be developing 1 reusable components library with web components.
